python/lsst/ts/m2/tcp_client.py

```diff
@@ -90,30 +98,39 @@ def __init__(
         self.timeout_in_second = timeout_in_second
 
         # Sequence ID generator
-        self._sequence_id_generator = salobj.index_generator()
+        if sequence_generator is None:
+            self._sequence_id_generator = salobj.index_generator()
+        else:
+            self._sequence_id_generator = sequence_generator
         self.last_sequence_id = -1
 
         self.queue = asyncio.Queue(maxsize=int(maxsize_queue))
 
         # Monitor loop task (asyncio.Future)
         self._monitor_loop_task = make_done_future()
 
-    async def connect(self, connect_retry_interval=1.0):
+    async def connect(self, connect_retry_interval=1.0, timeout=10.0):
         """Connect to the server.
 
         Parameters
         ----------
         connect_retry_interval : `float`, optional
             How long to wait before trying to reconnect when connection fails.
             (default is 1.0)
+        timeout : `float`, optional
+            Timeout in second. This value should be larger than the
+            connect_retry_interval. (default is 10.0)
 
-        Notes
-        -----
-        This will wait forever for a connection.
+        Raises
+        ------
+        asyncio.TimeoutError
+            Connection timeout.
```

Comment:
If you still want to raise `TimeoutError` (which I am not sure is exactly what you should do) consider expanding this to be more specific about when it is raised. See my comment in the code, but the "timeout" users specify is not actually a true timeout.

Reply:
Done in the commit: 320670f. Thanks!